main.tf

```diff
@@ -146,19 +146,42 @@ resource "google_service_networking_connection" "private_vpc_connection" {
 }
 
 resource "google_sql_database_instance" "gitlab_db" {
-  depends_on       = [google_service_networking_connection.private_vpc_connection]
-  name             = local.gitlab_db_name
-  region           = var.region
-  database_version = var.postgresql_version
+  depends_on           = [google_service_networking_connection.private_vpc_connection]
+  name                 = local.gitlab_db_name
+  region               = var.region
+  database_version     = var.postgresql_version
+  deletion_protection  = var.postgresql_del_protection
 
   settings {
-    tier            = var.postgresql_tier
-    disk_autoresize = true
+    tier               = var.postgresql_tier
+    availability_type  = var.postgresql_availability_type
+    disk_size          = var.postgresql_disk_size
+    disk_type          = var.postgresql_disk_type
+    disk_autoresize    = true
 
     ip_configuration {
       ipv4_enabled    = "false"
       private_network = google_compute_network.gitlab.self_link
     }
+
+    backup_configuration {
+      enabled                        = true
+      start_time                     = "02:00"
+      point_in_time_recovery_enabled = true
+    }
+
+    insights_config {
+      query_insights_enabled  = true
+      query_string_length     = 1024
+      record_application_tags = false
+      record_client_address   = true
+    }
```

Comment:
are you sure of this settings here ?

Reply:
Removed insight configuration. DB will be always queried by the same client.